### PR TITLE
DRILL-7276: Fixed an XSS vulnerability in Drill Web-UI query profile …

### PR DESCRIPTION
**Pull request by agozhiy**
…page

### CODE REVIEW

**Comment by vvysotskyi** [comment]
@agozhiy, Change looks good, thanks for fixing this issue.

Could you please revise other JS scripts and check the usage of `html(...)` method and other possibly dangerous methods which may cause similar issues?

**Comment by vvysotskyi** [comment]
@agozhiy, I have noticed another issue: when the script is inserted into the `autoLimit` filed, it breaks the warning window with the error message, could you please take a look at this issue?
image

**Comment by agozhiy** [comment]
@vvysotskyi thanks for finding this case, fixed it now. 
Cannot find any other dangerous usages of html() function, in most of the cases we use FreeMarker to substitute values.
Please, review.

File: exec/java-exec/src/main/resources/rest/alertModals.ftl [hunk 1]
```diff
@@ -62,14 +62,18 @@
...
+            updatedHtml=updatedHtml.replace(currKey, escapeHtml(inputValues[currKey]));
         }
         modalBody.innerHTML=updatedHtml;
       }
       //Show Alert
       $('#errorModal').modal('show');
     }
 
+    function escapeHtml(str) {
+        return str.replace(/&/g,'&amp;').replace(/</g,'&lt;').replace(/>/g,'&gt;') ;
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/resources/rest/alertModals.ftl]: Could you please escape other characters as it is described in [Cross Site Scripting Prevention Cheat Sheet](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.md#rule-1---html-escape-before-inserting-untrusted-data-into-html-element-content)?

Comment (agozhiy) [reply exec/java-exec/src/main/resources/rest/alertModals.ftl]: Done.

**Comment by agozhiy** [comment]
Done.

**Pull request closed without merging**